src/content/en/fundamentals/instant-and-offline/get-started/textcontent-3.md

```diff
@@ -0,0 +1,312 @@
+## Minified Code
+
+One simple and effective method is *minification*, which is essentially compressing a text 
+resource by removing its whitespace and unnecessary characters without changing its validity 
+or functionality. It doesn't sound that useful, but it is. For example, this little function 
+(part of a table-sorting script) initially contains 348 characters.
+
+```
+function sortables_init() {
+
+    // Find all tables with class sortable and make them sortable
+
+    if (!document.getElementsByTagName) return;
+
+    tbls = document.getElementsByTagName("table");
+
+    for (ti=0;ti<tbls.length;ti++) {
+
+   	 thisTbl = tbls[ti];
+
+   	 if (((' '+thisTbl.className+' ').indexOf("sortable") != -1) && (thisTbl.id)) {
+
+   		 ts_makeSortable(thisTbl);
+
+   	 }
+
+    }
+
+}
+```
+
+After minification, it looks like this, and only contains 257 characters. 
+
+```
+function sortables_init(){if(!document.getElementsByTagName)return;tbls=document.
+getElementsByTagName("table");for(ti=0;ti<tbls.length;ti++){thisTbl=tbls[ti];
+if(((''+thisTbl.className+'').indexOf("sortable")!=-1)&&(thisTbl.id)){ts_makeSortable(thisTbl)}}}
+```
+
+Sure, we can't read it now, but the browser still can. That's a 26% reduction in size and 
+required download time; pretty significant, even for this small sample.
+
+Looking at the bigger picture, the entire script block from which that snippet came is over 
+10k, but after minification it's down to 5,411 characters, a whopping 48% reduction.
+
+HTML and CSS can be minified in the same way, so that you can achieve improved load times for 
+both formatting- and behavior-related code.
+
+Many online and desktop minification tools are available; here are some examples.
+
+### Minifier
+
+This online tool minifies JavaScript or CSS via copy and paste. Paste the code, choose the 
+type, and click `Minify`.
+
+[http://www.minifier.org/](http://www.minifier.org/)
+
+![Minifier](images/image_300.png)
+
+### HTML Minifier
+
+Here's an online tool that also handles HTML, and automatically identifies the code type(s).
+
+[http://www.willpeavy.com/minifier/](http://www.willpeavy.com/minifier/)
+
+![HTML Minifier](images/image_301.png)
+
+### Kangax HTML Minifier
+
+This online tool offers a broad array of output customization options for the minified code.
+
+[https://kangax.github.io/html-minifier/](https://kangax.github.io/html-minifier/)
+
+![Kangax HTML Minifier](images/image_302.png)
+
+### NPM Packages
+
+The NPMJS site also has various minification packages, for Grunt
+
+[https://www.npmjs.com/package/grunt-html-minify](https://www.npmjs.com/package/grunt-html-minify)
+
+and Gulp,
+
+[https://www.npmjs.com/package/gulp-html-minifier](https://www.npmjs.com/package/gulp-html-minifier)
+
+and a popular package 
+
+[https://www.npmjs.com/package/html-minifier](https://www.npmjs.com/package/html-minifier)
+
+that includes a useful chart comparing its compression results with other methods.
+
+### Development vs. Deployment
```

Comment:
Suggestion to introduce separation of development and deployment concerns at the start of the doc.